File: frontend/src/lib/api/tlf.ts
import { apiClient } from "./client";

export interface TLFGenerationResponse {
  artifact_id: string;
  artifact_version_id: string;
  ai_decision_id: string;
  validation_run_id: string;
  table_count: number;
  study_id: string;
  source_adam_artifact_ids: string[];
}

export interface ListingFigureEntry {
  sap_section: string;
  output_title: string;
  output_type: "table" | "listing" | "figure";
  tlf_index: number;
  status: string;
}

export interface ListingFigureCatalog {
  sap_artifact_id: string | null;
  entries: ListingFigureEntry[];
}

export const tlfApi = {
  generateFromAdam: (adamArtifactId: string, token: string) =>
    apiClient.post<TLFGenerationResponse>(
      `/tlf/artifacts/${adamArtifactId}/generate-tlf`,
      { token }
    ),

  getCatalog: (artifactId: string, token: string) =>
    apiClient.get<ListingFigureCatalog>(`/tlf/artifacts/${artifactId}/catalog`, {
      token,
    }),
};

File: backend/app/schemas/tlf.py
"""Pydantic schemas for TLF generation API."""

from __future__ import annotations

from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class TLFGenerationResponse(BaseModel):
    artifact_id: UUID
    artifact_version_id: UUID
    ai_decision_id: UUID
    validation_run_id: UUID
    table_count: int
    study_id: UUID
    source_adam_artifact_ids: list[UUID] = Field(default_factory=list)


class ListingFigureEntry(BaseModel):
    sap_section: str
    output_title: str
    output_type: Literal["table", "listing", "figure"]
    tlf_index: int
    status: str = "programmed"


class ListingFigureCatalog(BaseModel):
    sap_artifact_id: UUID | None = None
    entries: list[ListingFigureEntry] = Field(default_factory=list)
